x/kvindexer/keeper/handler.go

```
@@ -101,3 +106,24 @@
 	return nil
 }
+
+func (k *Keeper) prune(ctx context.Context, height int64) {
+	if running := k.pruningRunning.Swap(true); running {
+		return
+	}
+
+	go func(ctx context.Context, height int64) {
+		defer k.pruningRunning.Store(false)
+
+		minHeight := height - int64(k.config.RetainHeight)
```

golangci-lint (1.62.2)

[high] G115: integer overflow conversion uint64 -> int64

(gosec)